sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -79,20 +79,25 @@ int coap_parse(coap_pkt_t *pkt, uint8_t *buf, size_t len)
     }
 
     /* token value (tkl bytes) */
-    if (coap_get_token_len(pkt)) {
+    if (coap_get_token_len(pkt) == 0) {
+        pkt->token = NULL;
+    }
+    else if (coap_get_token_len(pkt) <= COAP_TOKEN_LENGTH_MAX) {
         pkt->token = pkt_pos;
+        /* pkt_pos range is validated after options parsing loop below */
         pkt_pos += coap_get_token_len(pkt);
     }
     else {
-        pkt->token = NULL;
+        DEBUG("tkl invalid\n");
```

Comment:
Please prefix those DEBUG messages. Also, I would prefer to write out those abbreviations.

```suggestion
        DEBUG("nanocoap: token length invalid\n");
```